pyhelm/repo.py

```diff
@@ -29,14 +88,28 @@ def from_repo(repo_url, chart, version=None):
     if version is not None:
         versions = itertools.ifilter(lambda k: k['version'] == version,
                                      versions)
-
     try:
         metadata = sorted(versions, key=lambda x: x['version'])[0]
         for url in metadata['urls']:
             fname = url.split('/')[-1]
             try:
-                req = requests.get(url, stream=True)
-                fobj = cStringIO.StringIO(req.content)
+                if repo_scheme == 's3':
```

Comment:
This code block is very similar to the one in _repo_index_. Perhaps create a helper method that will contain the common functionality, and have this method and _repo_index_ only wrap it accordingly (e.g. wrap with yaml.load)